### ISSUE TITLE
MSTestRunner prioritizes VS2017 over VS2019

### ISSUE THREAD

**Issue by kcamp** [cake-build/cake#2752]
### What You Are Seeing?
When Visual Studio 2019 and 2017 are installed SxS, the `MSTestRunner` prioritizes the resolution of the lower version `MSTest.exe` binary.
The change made in [PR#2619](https://github.com/cake-build/cake/pull/2619/files) should probably have followed the pattern already in use there tool resolution as prioritized version-descending.

Working around by specifying an explicit `ToolPath` is convenient enough.
```
new MSTestSettings 
{
    ToolPath = new FilePath("C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/Common7/IDE/MSTest.exe")
}
```
It does, though, seem like this should be switched.

### What is Expected?
I would have expected the tool discovery to prioritize the highest installed version in the absence of any settings property providing affinity.

### What version of Cake are you using?
0.37.0

### Are you running on a 32 or 64 bit system?
x64

### What environment are you running on?  Windows? Linux? Mac?
Windows

### Are you running on a CI Server?  If so, which one?
N/A

### How Did You Get This To Happen? (Steps to Reproduce)
Attempt to run the `MSTest` alias against an `.orderedtest` file.  This would happen in any case where the resolution falls through to the `GetAlternativeToolPaths` method.

### Output Log
```
Executing task: Run-Smoke-Tests
Executing ordered test 'C:/git/Project/TestCases/bin/debug/TestCases.orderedtest'
Executing: "C:/Program Files (x86)/Microsoft Visual Studio/2017/Enterprise/Common7/IDE/mstest.exe" "/testcontainer:C:/git/Project/TestCases/bin/debug/TestCases.orderedtest" /noisolation
Microsoft (R) Test Execution Command Line Tool Version 15.0.27729.1
Copyright (c) Microsoft Corporation. All rights reserved.

Loading C:/git/Project/TestCases/bin/debug/TestCases.orderedtest...
Starting execution...

```

**Comment by kcamp** [comment]
I will have a PR momentarily.

**Comment by devlead** [comment]
fixed by #2753
